docs: add note for connecting debug console for old versions (#1843)
kata-containers/kata-containers: docs/Developer-Guide.md

This will pass `agent.debug_console agent.debug_console_vport=1026` to agent as kernel parameters, and sandboxes created using this parameters will start a shell in guest if new connection is accept from VSOCK.

> #### Start `kata-monitor` - OPTIONAL

Reviewer: ```suggestion
#### Start `kata-monitor` - ONLY NEEDED FOR 2.0.x
```

> For Kata Containers before `2.1-alpha1`, the `kata-runtime exec` command needs `kata-monitor` to get the sandbox's `vsock` address to connect to, first start `kata-monitor`.

Reviewer: ```suggestion
For Kata Containers `2.0.x` releases, the `kata-runtime exec` command depends on the`kata-monitor` running, in order to get the sandbox's `vsock` address to connect to.   Thus, first start the `kata-monitor` process.
```

```
$ sudo kata-monitor
```

`kata-monitor` will serve at `localhost:8090` by default.

> **Note:** Starting `kata-monitor` only required before Kata Containers before `2.1-alpha1 `.

Reviewer: ```suggestion
```

I'd remove the note entirely.
